src/N3Parser.js

```diff
@@ -1023,6 +1023,9 @@ export default class N3Parser {
     this._inversePredicate = false;
     this._quantified = Object.create(null);
 
+    // If the parser is requested to provide comments through a callback, override the current lexer’s configuration parameter to emit comments
+    if (commentCallback) this._lexer._comments = true;
```

Comment:
Shouldn't set private (`_`) fields of others; need to set this up differently (constructor).

Reply:
In the constructor might not be the right place though: only when the parse function has a commentcallback you want to enable the comments in the lexer. I would propose a setter then instead. Do you agree?

Reply:
Setter is good 👍

Reply:
Alternatively, pass `comment` option to the constructor, which then initiates monkey patching (see below). And then `this._commentCallback` is just set to `() => null` if no `commentCallback` is passed.

Reply:
I am afraid developers will find it counter-intuitive that they must add a callback, _and must_ also have constructed the parser in a different way or it won’t work. I think the setter is the best solution here.

Reply:
Implemented the setter